OverlayPlugin.Core/NetworkProcessors/LineActorControlExtra.cs

```diff
@@ -0,0 +1,164 @@
+                switch (region)
+                {
+                    case GameRegion.Global:
+                        {
+                            actorControlTypeStr = "Machina.FFXIV.Headers.Server_ActorControl";
+                            break;
+                        }
+                    case GameRegion.Korean:
+                        {
+                            actorControlTypeStr = "Machina.FFXIV.Headers.Korean.Server_ActorControl";
+                            break;
+                        }
+                    case GameRegion.Chinese:
+                        {
+                            actorControlTypeStr = "Machina.FFXIV.Headers.Chinese.Server_ActorControl";
+                            break;
+                        }
+                    default:
+                        {
+                            return;
+                        }
+                }
+
+                Type actorControlType = mach.GetType(actorControlTypeStr);
+                RegionalizedInfo info = new RegionalizedInfo(headerType, actorControlType, netHelper);
+                regionalized = info;
+            }
+            catch (System.IO.FileNotFoundException)
+            {
+                logger.Log(LogLevel.Error, Resources.NetworkParserNoFfxiv);
+            }
+            catch (Exception e)
+            {
+                logger.Log(LogLevel.Error, Resources.NetworkParserInitException, e);
+            }
+        }
+
+        private unsafe void MessageReceived(string id, long epoch, byte[] message)
+        {
+            RegionalizedInfo info = regionalized;
+            if (info == null)
+                return;
+
+            if (message.Length < info.packetSize)
+                return;
+
+            fixed (byte* buffer = message)
+            {
+                if (*(ushort*)&buffer[info.offsetMessageType] == info.packetOpcode)
+                {
+                    object header = Marshal.PtrToStructure(new IntPtr(buffer), info.headerType);
+                    UInt32 sourceId = (UInt32)info.fieldCastSourceId.GetValue(header);
+
+                    object packet = Marshal.PtrToStructure(new IntPtr(buffer), info.actorControlType);
+                    UInt16 category = (UInt16)info.fieldCategory.GetValue(packet);
+
+                    if (Enum.IsDefined(typeof(Server_ActorControlCategory), category))
+                    {
+                        UInt32 param1 = (UInt32)info.fieldParam1.GetValue(packet);
+                        UInt32 param2 = (UInt32)info.fieldParam2.GetValue(packet);
+                        UInt32 param3 = (UInt32)info.fieldParam3.GetValue(packet);
+                        UInt32 param4 = (UInt32)info.fieldParam4.GetValue(packet);
+
+                        string line = string.Format(CultureInfo.InvariantCulture,
+                            "{0:X8}|{1:X4}|{2:X8}|{3:X8}|{4:X8}|{5:X8}",
+                            sourceId, category, param1, param2, param3, param4);
```

Comment:
Could we do param1-param4 as variable length output?  It looks like `SetAnimationState` has a pretty confined set of values.

Also curious about `param1` and `param2`, since `SpawnNpcExtra` refers to the byte as `animationState`.  Is `param1` equivalent to `animationState`, and `param2` tells the client to display the animation?  Or do we not have enough data points to infer that?

(And related, I suppose -- are you keeping the names generic because we may add additional `ActorControlExtra` categories in the future?)

Reply:
quisquous did the initial writeup of this code so I can't speak to their original intentions, but I would prefer to keep the fields generic to allow other `ActorControl` categories later. Refer to the other `ActorControl` categories I mention in #276 for more potential options that we might allow through in the future.

I'm fine with changing the output format to variable length if that makes sense.

Reply:
> quisquous did the initial writeup of this code so I can't speak to their original intentions, but I would prefer to keep the fields generic to allow other `ActorControl` categories later. Refer to the other `ActorControl` categories I mention in #276 for more potential options that we might allow through in the future.

Yep, makes sense.

> I'm fine with changing the output format to variable length if that makes sense.

I don't feel strongly about it.  My reason for suggesting it is purely cosmetic, and ymmv - it's more work when looking through a log file and long strings of `0`s to notice the lone `1`, and slightly cleaner trigger-writing to look for `1` instead of `0000001`.  But I'm fine with whatever you prefer.